pennylane_lightning/lightning_tensor/lightning_tensor.py

```diff
@@ -0,0 +1,343 @@
+    return method in _methods
+
+
+@simulator_tracking
+@single_tape_support
+class LightningTensor(Device):
+    """PennyLane Lightning Tensor device.
+
+    A device to perform tensor network operations on a quantum circuit.
+
+    Args:
+        wires (int): The number of wires to initialize the device with.
+            Defaults to ``None`` if not specified.
+        backend (str): Supported backend. Currently, only ``quimb`` is supported.
+        method (str): Supported method. Currently, only ``mps`` is supported.
+        shots (int): How many times the circuit should be evaluated (or sampled) to estimate
+            the expectation values. Currently, it can only be ``None``, so that computation of
+            statistics like expectation values and variances is performed analytically.
+        c_dtype: Datatypes for the tensor representation. Must be one of
+            ``np.complex64`` or ``np.complex128``.
+        **kwargs: keyword arguments. The following options are currently supported:
+
+            ``max_bond_dim`` (int): Maximum bond dimension for the MPS simulator.
+                It corresponds to the number of Schmidt coefficients retained at the end of the SVD algorithm when applying gates. Default is ``None``.
+            ``cutoff`` (float): Truncation threshold for the Schmidt coefficients in a MPS simulator. Default is ``1e-16``.
+    """
+
+    # pylint: disable=too-many-instance-attributes
+
+    # So far we just consider the options for MPS simulator
+    _device_options = (
+        "backend",
+        "c_dtype",
+        "cutoff",
+        "method",
+        "max_bond_dim",
+    )
+
+    _new_API = True
+
+    # pylint: disable=too-many-arguments
+    def __init__(
+        self,
+        *,
+        wires=None,
+        backend="quimb",
+        method="mps",
+        shots=None,
+        c_dtype=np.complex128,
+        **kwargs,
+    ):
+
+        if not accepted_backends(backend):
+            raise ValueError(f"Unsupported backend: {backend}")
+
+        if not accepted_methods(method):
+            raise ValueError(f"Unsupported method: {method}")
+
+        if shots is not None:
+            raise ValueError("LightningTensor does not support finite shots.")
+
+        super().__init__(wires=wires, shots=shots)
+
+        self._num_wires = len(self.wires) if self.wires else 0
+        self._backend = backend
+        self._method = method
+        self._c_dtype = c_dtype
+
+        # options for MPS
+        self._max_bond_dim = kwargs.get("max_bond_dim", None)
+        self._cutoff = kwargs.get("cutoff", 1e-16)
+
+        self._interface = None
+        interface_opts = self._setup_execution_config().device_options
+
+        if self.backend == "quimb" and self.method == "mps":
+            self._interface = QuimbMPS(
+                self._num_wires,
+                interface_opts,
+                self._c_dtype,
+            )
```

Comment:
We should raise a runtime error here as the else of this branch, since no other backend is supported.

Reply:
I agree, but I haven't raised a runtime error since `QuimbMPS` is the only option available if we get to this point (notice the checks on `accepted_backends` and `accepted_methods`). I just inserted the condition `if self.backend == "quimb" and self.method == "mps":` for clarity and as a reminder that we'll have more backends soon. At this point, I think every precaution that takes into account to possibility of having multiple backends is not strictly necessary, as we have only one, but maybe it is worth doing that for clarity

Reply:
I'd rather we put this in for clarity, even with the coming second PR (and the CUDA one after). Any unsupported option is something we should trigger a failure for.